lib/fs-provider.js

```diff
@@ -0,0 +1,98 @@
+'use strict'
+const fs = require('fs')
+const fse = require('fs-extra')
+const join = require('path').join
+
+module.exports = function fsProvider (config) {
+  let serialize = config && !!config.serialize
+  let directory = (config && config.directory) || join(__dirname, '.persist')
+  fse.mkdirpSync(directory)
+
+  let provider = {
+    type: 'fs',
+
+    get: function (key, cb) {
+      key = encodeURIComponent(key)
+
+      fs.readFile(join(directory, key), { encoding: 'utf8' }, function (err, value) {
+        if (err && err.code === 'ENOENT') {
+          return cb(null)
+        }
+        if (err) {
+          return cb(err)
+        }
+        value = (serialize) ? JSON.parse(value) : value
+        cb(null, value)
+      })
+    },
+
+    mget: function (keys, cb) {
+      keys = keys || []
+      let total = 0
+      let values = []
+      let anyError = null
+
+      let gather = (err, idx, val) => {
+        anyError = anyError || err
+        values[idx] = val
+        total++
+        if (keys.length === total) {
```

Comment:
Was trying not to add any new dependencies. `fs-extra` made sense but didn't want to pull in the rest of the world